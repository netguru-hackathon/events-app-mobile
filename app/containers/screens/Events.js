import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  FlatList,
  Image,
  StyleSheet,
  View,
} from 'react-native';
import I18n from '../../utils/translations';
import { fetchEvents } from '../../actions/events';
import { EventsItem } from './EventsItem';
import { RenderActivityIndicator } from '../shared/RenderActivityIndicator';
import colors from '../../constants/colors';

const styles = StyleSheet.create({
  icon: {
    width: 20,
    height: 20,
  },
  separator: {
    height: 5,
    marginLeft: 15,
    marginRight: 15,
    backgroundColor: colors.TRANSPARENT,
  },
  view: {
    marginTop: 5,
  },
});

class Events extends Component {
  static propTypes = {
    fetchEvents: PropTypes.func.isRequired,
    isStarted: PropTypes.bool.isRequired,
    isFetching: PropTypes.bool.isRequired,
    events: PropTypes.arrayOf(PropTypes.object).isRequired,
    navigation: PropTypes.shape({
      state: PropTypes.object.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  static navigationOptions = {
    tabBarLabel: I18n.t('events'),
    showIcon: true,
    tabBarIcon: ({ tintColor }) => (
      <Image
        source={require('../../img/EventsIcon.png')}
        style={[styles.icon, { tintColor }]}
      />
    ),
    headerTitle: I18n.t('events'),
    headerTintColor: colors.WHITE,
    headerStyle: {
      backgroundColor: colors.BLACK_OPACITY,
    },
  };

  componentWillMount() {
    this.props.fetchEvents();
  }

  // eslint-disable-next-line
  renderEvent({ item }) {
    const { navigation } = this.props;

    return (
      <EventsItem
        key={`event${item.id}`}
        event={item}
        onPress={() => navigation.navigate('Event', { item })}
      />
    );
  }

  // eslint-disable-next-line
  renderSeparator() {
    return (
      <View style={styles.separator} />
    );
  }

  renderEventsList() {
    const { isStarted, isFetching, events } = this.props;

    if (isStarted && isFetching) return <RenderActivityIndicator />;
    return (
      <FlatList
        /* https://github.com/facebook/react-native/issues/13316 */
        removeClippedSubviews={false}
        data={events}
        keyExtractor={item => item.id}
        ItemSeparatorComponent={this.renderSeparator}
        // eslint-disable-next-line
        renderItem={this.renderEvent.bind(this)}
      />
    );
  }

  render() {
    return (
      <View style={styles.view}>
        {this.renderEventsList()}
      </View>
    );
  }
}

const mapStateToProps = state => ({
  isStarted: state.events.isStarted,
  isFetching: state.events.isFetching,
  events: state.events.items,
});

const mapDispatchToProps = {
  fetchEvents,
};

export default connect(mapStateToProps, mapDispatchToProps)(Events);
