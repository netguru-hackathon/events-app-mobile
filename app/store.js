import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';

import axiosClient from './api/client';
import Reducers from './reducers/index';

const store = applyMiddleware(
  thunk,
  axiosMiddleware(axiosClient, { returnRejectedPromiseOnError: true }),
)(createStore)(Reducers);

export default store;
