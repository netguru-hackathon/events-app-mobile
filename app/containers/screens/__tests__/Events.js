import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Events from '../Events';
import store from '../../../store';

test('renders correctly', () => {
  const tree = renderer.create(
    <Events
      store={store}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
