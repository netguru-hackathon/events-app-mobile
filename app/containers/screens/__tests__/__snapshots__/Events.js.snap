// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Events
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Loading..
  </Text>
</View>
`;
